src/parser/parser.cc

```diff
@@ -1231,14 +1335,38 @@ class Parser {
           }
         }
         default: {
-          std::stringstream msg;
-          msg << "expected an expression found  " << Pretty(next->token_type);
-          diag_ctx.Emit({next->line, next->column, msg.str()});
-          diag_ctx.Render(std::cout);
+          this->diag_ctx->EmitFatal(DiagnosticBuilder(DiagnosticLevel::Error, next->span)
+                                    << "expected an expression found  "
+                                    << Pretty(next->token_type));
           return Expr();
         }
       }
     });
+
+    if (WhenMatch(TokenType::Period)) {
+      auto index = Match(TokenType::Integer).ToNumber();
+      expr = relay::TupleGetItem(expr, index);
+    }
+
+    return expr;
+  }
+
+  /*! \brief Parse a hierarchical name. */
+  Array<String> ParseHierName() {
```

Comment:
Maybe expand this to ParseHierarchicalName? It's not a common shortening, took a while to figure out what the function name meant when I saw it in code.

Reply:
Yeah I will document this, was quickly hacking up with Jason on the stream in order to unblock, worth coming back to it.